src/main/java/com/redhat/devtools/lsp4ij/features/formatting/LSPEditorImprovementsTypedHandler.java

```diff
@@ -126,6 +129,20 @@ private static String getStringLiteral(@NotNull PsiFile file,
                                            int offset) {
         Set<Character> quoteCharacters = LSPIJEditorUtils.getQuoteCharacters(file);
         if (!ContainerUtil.isEmpty(quoteCharacters)) {
+            // Try to determine this using semantic tokens
```

Comment:
If the language server provides string literal semantic tokens, we can use that to get the string literal without having to search for it textually.